### PR TITLE
[codex] Fix TUI protocol views and tool handling

### PR DESCRIPTION
**Pull request by lsm**
## Summary

- Fix OpenAI Responses API tool-call handling and provider protocol bridging for agent/tool execution.
- Add TUI view modes and selectors for `/view`, `/perm`, `/think`, and `/sessions` style choices.
- Render protocol events, tool/agent errors, reasoning, and activity according to the selected transcript view.
- Improve composer input behavior: block cursor, selection-safe mouse scroll, Shift+Enter newline, model persistence, and default bypass permissions.
- Vendor Zigzag so Makai can use normal mouse tracking (`1000` + SGR `1006`) instead of all-motion tracking (`1003`), preserving terminal drag selection while keeping wheel scrolling.

## Root Cause

The OpenAI Responses stream and TUI transcript handling were missing several event/tool edge cases. Separately, enabling mouse support through Zigzag used all-motion tracking, which captured drag selection and prevented terminal-native selection.

## Validation

- `zig/vendor/zigzag`: `zig build test --summary all` passed (`110/110`).
- `zig`: `zig build --summary all` passed.
- `zig`: `zig build test-unit-tui --test-timeout 10s --summary all` reached `299/300`; one pre-existing timeout remains: `file.test.file read caps hashed output expansion`.
- `git diff --check` passed.

### CODE REVIEW

**Comment by lsm** [comment]
@codex review

**Comment by lsm** [comment]
# TUI Feature Gap Analysis

## 1. Unimplemented AppModes (6 of 10)

`state.zig` declares 10 modes. Only 4 have views.

| Mode | Has View |
|------|----------|
| `normal` | Yes |
| `approval` | Yes |
| `preview` | Yes |
| `session_picker` | Yes |
| `model_picker` | **No** |
| `login_picker` | **No** |
| `permission_picker` | **No** |
| `view_picker` | **No** |
| `thinking_picker` | **No** |
| `login_input` | **No** |

`TuiModel.update` only routes `approval` and `session_picker`. Remaining 6 modes are dead declarations.

## 2. Stub Commands

| Command | Status |
|---------|--------|
| `/compact` | Returns `"compact not yet implemented"` |
| `/resume [id]` | Returns `"resume by id is not wired yet"` |
| `/diff` | Heuristic only — name-contains-"edit", no git diff |

## 3. Missing Slash Commands

- `/abort` — cancel current turn mid-stream
- `/edit` — edit last message
- `/copy` — copy response to clipboard
- `/export` — export session transcript
- `/redo` — re-run last turn

## 4. Remote Runtime Gaps

| Gap | Severity |
|-----|----------|
| SSE/WebSocket transport returns `UnsupportedRemoteTransport` | Stub |
| Remote steering returns `RemoteSteeringUnsupported` | Missing |
| Single reconnect, no backoff | Weak |
| Blocking stdio pump — no async yield | Missing |
| `tool_protocol_override_fn` never populated | Dead field |
| `TuiRemoteConfig.endpoint` rejected if non-empty | Dead field |
| No graceful shutdown timeout | Missing |
| No backpressure — drops oldest silently | Missing |
| MCP bind but no unbind in deinit | Resource leak |

## 5. Config-View Disconnect

`vim_mode` and `caveman_mode` declared in `ModeSettings` but composer hints render identical text regardless. No vim keybinding handling in `TuiModel.update`.

## 6. Session Management

- No session create/delete in picker
- No session fork/branch
- No session search/filter
- Remote resume unsupported

## 7. Tool Panel

- Hashline preview hardcodes `"hashline_edit"` tool name only
- Artifact preview kind uses same diff styling
- No tool retry/resubmit
- No copy-to-clipboard on output

## 8. Test Coverage Gaps

| Area | Status |
|------|--------|
| 8 of 12 command handlers | Untested |
| Session layer methods | Only 1 test (deinit) |
| Async tool with on_update | Untested |
| Concurrency stress | Untested |
| Vim mode | No code, no tests |
| Model picker flow | No code, no tests |

## 9. UX Polish

- No markdown rendering in transcript
- Timestamps on entries exist but not rendered
- No model switching UI
- No keyboard shortcut reference outside welcome text
- Composer doesn't show cursor position or multi-line indicator

---

### Priority Ranking

**P0 (Blocks shipping):**
1. Model picker view — users can't switch models
2. `/compact` stub — returns error to user
3. SSE/WebSocket remote transport stub
4. Remote steering unsupported

**P1 (Should fix):**
5. Session create/delete in picker
6. `/abort` command for cancel mid-stream
7. `/edit` for edit last message
8. Backpressure signaling in remote pump
9. MCP unbind in deinit

**P2 (Nice to have):**
10. Login/permission picker views
11. Vim mode keybindings
12. Copy to clipboard
13. Markdown rendering in transcript
14. Session search/filter
15. Timestamps in transcript

File: zig/src/tui/runtime.zig [hunk 1]
```diff
@@ -108,12 +114,41 @@ pub const TuiRuntimeOptions = struct {
     tools: []const agent.AgentTool = &.{},
     mcp_config_json: ?[]const u8 = null,
     permission_engine: ?*permission.PermissionEngine = null,
+    workspace_root: []const u8 = "",
     tool_approval_ctx: ?*anyopaque = null,
     tool_approval_callback: ?ToolApprovalCallback = null,
+    permission_mode: PermissionMode = .bypass,
```
[the remark this answers is not in the document]
Comment (lsm) [reply zig/src/tui/runtime.zig:121]: Keeping bypass as the default is intentional for this PR. The TUI still exposes /permissions for users who want ask mode, but the product requirement for this branch is to start in bypass mode by default.
